scripts/msys2-install-dependencies.sh

```diff
@@ -1,59 +1,72 @@
 #!/usr/bin/env bash
+set -e
 
 QT="$1"
 
-if [ -z $MSYSTEM ]; then
-  MSYSTEM_DEFAULT=UCRT64
-  # Possible values: (MSYS|UCRT64|CLANG64|CLANGARM64|CLANG32|MINGW64|MINGW32)
-  # For explanation of options see: https://www.msys2.org/docs/environments/
-  echo "MSYSTEM is unset or blank, defaulting to '${MSYSTEM_DEFAULT}'";
-  export MSYSTEM=$MSYSTEM_DEFAULT
-else
-  echo "MSYSTEM is set to '$MSYSTEM'";
-fi
-
 date "+### %Y-%m-%d %T msys2-install-dependencies started"
 
-pacman --query --explicit
+if [[ -z "${GITHUB_RUN_ID}" ]]; then
```

Comment:
Please use a dedicated flag instead of indirectly relying on a magic runtime variable. It can be just another environment variable like `DRY_RUN` or `PRINT_PACKAGES_ONLY` or a dedicated command line flag.

Reply:
Sorry, I did not include a reason, why I'm suggesting that. The idea is that this feature of only listing the packages might be useful in other scenarios too, like people building their own images or running on a different CI. In those cases it would be confusing if that requires to set a variable called GITHUB_RUN_ID.

Reply:
Recommended by who? If that is Github, I think it's not a valid point in this specific case, but just self-interest, but I'm happy to read up on that if you have a pointer. Right now I see no reason to put Github specifics into the script if it's mostly trivial to have the dependency in the workflow xml instead.

Reply:
Ok, let's do that :grin: